### PR TITLE
Increase string size limit imposed by WebPA when SETting a parameter

### PR DESCRIPTION
**Pull request by Sreedevi29**

### CODE REVIEW

File: source/broadband/webpa_adapter.c [hunk 1]
```diff
@@ -607,10 +607,27 @@ WDMP_STATUS validate_parameter(param_t *param, int paramCount, REQ_TYPE type)
                         return WDMP_ERR_INVALID_PARAM;
                 }
 
-                if(strlen(param[i].value) >= MAX_PARAMETERVALUE_LEN)
+                //Increase parameter value to 200k for T2 ReportProfiles
+		if(strcmp(param[i].name, "Device.X_RDKCENTRAL-COM_T2.ReportProfiles") == 0) 
                 {
-                        return WDMP_ERR_INVALID_PARAM;
+			WalInfo("----SET/SET-ATTRIBUTES for T2 report profile-----\n");
```
Comment (sadhyama) [inline source/broadband/webpa_adapter.c:613]: Remove this 2 WalInfo logs , retain WalError log for failure case when >200K.

File: source/broadband/webpa_adapter.c [hunk 2]
```diff
@@ -607,10 +607,27 @@ WDMP_STATUS validate_parameter(param_t *param, int paramCount, REQ_TYPE type)
...
+			WalInfo("length of parameter value %ld \n",strlen(param[i].value));
+                        if(strlen(param[i].value) >= MAX_TELEMETRY_PARAMETERVALUE_LEN)
+			{
+			WalError("T2 reportProfiles SET/SET-ATTRIBUTES is not supported as string size is more than 200k\n");
+			return WDMP_ERR_INVALID_PARAM;
+			}
```
Comment (sadhyama) [inline source/broadband/webpa_adapter.c:619]: correct the tab spacing for this if block.

File: source/broadband/webpa_adapter.c [hunk 3]
```diff
@@ -607,10 +607,27 @@ WDMP_STATUS validate_parameter(param_t *param, int paramCount, REQ_TYPE type)
...
                 }
+		else
+		{
+			WalInfo("----SET/SET-ATTRIBUTES for Other than T2-----\n");
+			WalInfo("length of parameter value %ld \n",strlen(param[i].value));
```
Comment (sadhyama) [inline source/broadband/webpa_adapter.c:624]: Same as above, remove this WalInfo logs which is for debugging purpose

File: source/broadband/webpa_adapter.c [hunk 4]
```diff
@@ -607,10 +607,27 @@ WDMP_STATUS validate_parameter(param_t *param, int paramCount, REQ_TYPE type)
...
+                	if(strlen(param[i].value) >= MAX_PARAMETERVALUE_LEN)
+                	{
+			WalError("SET/SET-ATTRIBUTES is not supported as string size is more than 4k \n");
```
Comment (sadhyama) [inline source/broadband/webpa_adapter.c:627]: I think we can avoid this WalError and keep it same as existing code for other parameters >4K.
correct tab spacing .

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: source/broadband/webpa_adapter.c (hunk 3 above)
Comment (Sreedevi29) [reply source/broadband/webpa_adapter.c]: Incorporated

File: source/broadband/webpa_adapter.c (hunk 4 above)
Comment (Sreedevi29) [reply source/broadband/webpa_adapter.c]: Incorporated

**Review by sadhyama — APPROVED** [review]

**Review by sadhyama — APPROVED** [review]

**Pull request merged**